src/mpid/ch4/netmod/ofi/ofi_am_events.h

```diff
@@ -341,8 +341,7 @@ static inline int MPIDI_OFI_handle_lmt_ack(MPIDI_OFI_am_header_t * msg_hdr)
         MPIDI_OFI_rma_key_unpack(mr_key, NULL, &key_type, &idx_back);
         MPIR_Assert(MPIDI_OFI_KEY_TYPE_HUGE_RMA == key_type);
 
-        MPIDI_OFI_index_allocator_free(MPIDI_OFI_COMM(MPIR_Process.comm_world).rma_id_allocator,
-                                       (uint64_t) idx_back);
+        MPIDI_OFI_index_allocator_free(MPIDI_Global.mr_key_allocator, (uint64_t) idx_back);
```

Comment:
Will we have multiple allocators at some point? I'm wondering if passing this global var through the API is really necessary.

Reply:
I think the index allocator is a general implementation that does not tie to MR key itself. Therefore, it is better to keep passing this global allocator and keep the implementation reusable.

Reply:
That sounds like an overgeneralization, which should be avoided.  If we really need it later, we can always create a generic version.

Reply:
I have removed the passing of the allocator pointer. Now it is a global variable in specific to the mr_key allocator.